projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -106,6 +103,15 @@ public AutorecoveringConnection(ConnectionFactory factory, string clientProvided
             this.ClientProvidedName = clientProvidedName;
         }
 
+        /// <summary>
+        /// Finalizes an instance of the <see cref="AutorecoveringConnection"/> class.
+        /// </summary>
+        ~AutorecoveringConnection()
```

Comment:
There is no reason to implement finalizers on any of these types. This is only something you need to do if you have unmanaged resources that need have to be freed even when `Dispose` hasn't been called.

Reply:
@bording is correct, and [the provided link backs it up](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern#finalizable-types).

Reply:
How come? Doesn't `CreateConnection` create a `Socket` instance?

Of course this is broken because `IProtocol` doesn't implement `IDisposable`. But we have to start somewhere.

Reply:
`Socket` is still a managed resource, so just call `Dispose`, and it will handle any native OS cleanup.

You only need a finalizer when you have directly allocated native resources that won't be freed if you don't do it yourself.

Reply:
@bording 

> You only need a finalizer when you have directly allocated native resources that won't be freed if you don't do it yourself.

Fair enough!